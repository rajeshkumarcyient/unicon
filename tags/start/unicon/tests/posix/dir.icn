
$include "posix.icn"

procedure main()

   mkdir("./tmp", 8r0755) | stop("Couldn't make ./tmp: ", sys_errstr(&errno))

   f := stat("tmp") | stop("Couldn't stat: ", sys_errstr(&errno))

   write(format(f, "tmp"))
   
   rmdir("tmp") | stop("Couldn't remove ./tmp: ", sys_errstr(&errno))

   f := stat("tmp") | write("Couldn't stat: ", sys_errstr(&errno))

end

procedure format(p, name)

   s := right(p.blocks, 4) || " " || p.mode || " " ||
      right(p.nlink, 3) || " "  ||
      " " || right(p.size, 8) || " " || name

   if p.mode[1] == "l" then
      s ||:= " -> " || readlink(name)

   return s
end

