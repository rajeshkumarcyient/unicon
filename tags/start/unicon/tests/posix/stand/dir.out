   8 drwxr-xr-x   2      4096 tmp
Couldn't stat: No such file or directory
